src/integritysetup/integrity-util.c

```diff
@@ -34,9 +34,24 @@ int parse_integrity_options(
                 else if (streq(word, "allow-discards")) {
                         if (ret_activate_flags)
                                 *ret_activate_flags |= CRYPT_ACTIVATE_ALLOW_DISCARDS;
-                } else if (streq(word, "no-journal")) {
-                        if (ret_activate_flags)
-                                *ret_activate_flags |= CRYPT_ACTIVATE_NO_JOURNAL;
+                } else if ((val = startswith(word, "mode="))) {
+                        if (streq(val, "journal")) {
+                                if (ret_activate_flags) {
+                                        *ret_activate_flags &= ~CRYPT_ACTIVATE_NO_JOURNAL;
+                                        *ret_activate_flags &= ~CRYPT_ACTIVATE_NO_JOURNAL_BITMAP;
+                                }
```

Comment:
merge the two lines?

```c
if (ret_activate_flags)
        *ret_activate_flags &= ~(CRYPT_ACTIVATE_NO_JOURNAL|CRYPT_ACTIVATE_NO_JOURNAL_BITMAP);
```

Reply:
I kinda like the symmetry of the stanzas in the original version, but I'll concede that your version is slightly more efficient. Will fix.